app/controllers/user_controller.rb

```diff
@@ -1,13 +1,51 @@
+require_relative '../../models/user'
+require_relative './application_controller'
+require 'bcrypt'
+
 class UserController < ApplicationController
-  get '/users' do
-    @users = User.all
-    haml :users
+  get '/signup' do
+    haml :'user_management/signup'
+  end
+
+  get '/login' do
+    haml :'user_management/login'
   end
 
-  post '/submit' do
-    @user = User.new(params[:user])
+  get '/logout' do
+    warden_handler.logout
+    flash[:success] = 'Successfully logged out'
+    redirect '/'
+  end
+
+  post '/signup' do
+    if /^(?=.*[a-zA-Z])(?=.*[0-9])(?=.*[\W]).{8,}$/.match?(params[:password])
+      @user = User.new(first_name: params[:first_name],
```

Comment:
I'd suggest that you have a private method that extracts user params. That way, we reduce the work being done by this route handler. If you do that, you can simply have:
```ruby
@user = User.new(user_params)
```